src/utils.ts

```diff
@@ -435,3 +444,17 @@ export const hasNestedMappings = (node: TypeName): boolean => {
 
   return result;
 };
+
+/**
+ * Extracts the fields of a struct
+ * @dev returns the fields names of the struct as a string array
+ * @param node The struct to extract the fields from
+ * @returns The fields names of the struct
+ */
+export const extractStructFields = (node: TypeName): string[] | null => {
+  const fields = getStructFields(node);
+
+  if (!fields.length) return null;
+
+  return fields.map((field) => (field as VariableDeclaration).name).filter((name) => name);
+};
```

Comment:
2 small improvements to this function:
- we can try to avoid confusion between `extractStructFields` and `getStructFields` by naming the former something like `extractFieldNames`
- let's return `[]` in case the `fields` variable is empty for consistency (the other function does that) and to prevent null checks in code